lib/App/DuckPAN/Restart.pm

```diff
@@ -0,0 +1,98 @@
+package App::DuckPAN::Restart;
+# ABSTRACT: Automatic restarting of application on file change
+
+use File::Find::Rule;
+use Filesys::Notify::Simple;
+
+use strict;
+
+use Moo::Role;
+
+requires '_run_app';
+
+sub run_restarter {
+    my ($self, $args) = @_;
+
+    # will keep (re)starting the server until user ctrl-c
+    while(1){
+        defined(my $app = fork) or die 'Failed to fork application';
+        unless($app){ # app kid
+            $self->_run_app($args);
+            exit 0;
+        }
+
```

Comment:
Exactly.  We need to clean up the spawned app before exiting with the second fork.  If we want the structure to be the same, we can change the first fork to the latter's.